File: leetcode/1624. Largest Substring Between Two Equal Characters/main.py
class Solution:
    def maxLengthBetweenEqualCharacters(self, s: str) -> int:
        max_length = -1

        for left in range(len(s)):
            for right in range(left + 1, len(s)):
                if s[left] == s[right]:
                    max_length = max(max_length, right - left - 1)

        return max_length

File: leetcode/1624. Largest Substring Between Two Equal Characters/main.go
package main

func maxLengthBetweenEqualCharacters(s string) int {
	maxLength := -1

	for left := 0; left < len(s); left++ {
		for right := left + 1; right < len(s); right++ {
			if s[left] == s[right] {
				maxLength = max(maxLength, right-left-1)
			}
		}
	}

	return maxLength
}
